toolkit/cli/smart-contracts-commands/src/register.rs

```diff
@@ -0,0 +1,58 @@
+use jsonrpsee::http_client::HttpClient;
+use partner_chains_cardano_offchain::{await_tx::FixedDelayRetries, register::run_register};
+use sidechain_domain::{
+	AdaBasedStaking, AuraPublicKey, CandidateRegistration, GrandpaPublicKey, MainchainPublicKey,
+	MainchainSignature, SidechainPublicKey, SidechainSignature, UtxoId,
+};
+
+use crate::{parse_sidechain_public_keys, read_private_key_from_file};
+
+#[derive(Clone, Debug, clap::Parser)]
+pub struct RegisterCmd {
+	#[clap(flatten)]
+	common_arguments: crate::CommonArguments,
+	#[arg(long)]
+	genesis_utxo: UtxoId,
+	#[arg(long)]
+	registration_utxo: UtxoId,
+	#[arg(long)]
+	payment_key_file: String,
+	#[arg(long, value_name = "SIDECHAIN_KEY:AURA_KEY:GRANDPA_KEY", value_parser = parse_sidechain_public_keys)]
+	sidechain_public_keys: (SidechainPublicKey, AuraPublicKey, GrandpaPublicKey),
+	#[arg(long)]
+	sidechain_signature: SidechainSignature,
```

Comment:
I'm not sure what to do about using `sidechain`. We are "partner chains" for quite a long time but we have not renamed it in many places.